plugins/wpgraphql-logging/src/Admin/Settings_Page.php

```diff
@@ -0,0 +1,212 @@
+		add_action( 'admin_menu', [ $this, 'register_settings_page' ], 10, 0 );
+		add_action( 'admin_init', [ $this, 'register_settings_fields' ], 10, 0 );
+		add_action( 'admin_enqueue_scripts', [ $this, 'load_scripts_styles' ], 10, 1 );
+	}
+
+	/**
+	 * Initialize the field collection.
+	 */
+	public function init_field_collection(): void {
+		$this->field_collection = new Settings_Field_Collection();
+	}
+
+	/**
+	 * Registers the settings page.
+	 */
+	public function register_settings_page(): void {
+		if ( is_null( $this->field_collection ) ) {
+			return;
+		}
+
+		$tabs = $this->field_collection->get_tabs();
+
+		$tab_labels = [];
+		foreach ( $tabs as $tab_key => $tab ) {
+			if ( ! is_a( $tab, Settings_Tab_Interface::class ) ) {
+				continue;
+			}
+
+			$tab_labels[ $tab_key ] = $tab->get_label();
+		}
+
+		$page = new Menu_Page(
+			__( 'WPGraphQL Logging Settings', 'wpgraphql-logging' ),
+			'WPGraphQL Logging',
+			self::PLUGIN_MENU_SLUG,
+			trailingslashit( WPGRAPHQL_LOGGING_PLUGIN_DIR ) . 'src/Admin/Settings/Templates/admin.php',
+			[
+				'wpgraphql_logging_main_page_config' => [
+					'tabs'        => $tab_labels,
+					'current_tab' => $this->get_current_tab(),
+				],
+			],
+		);
+
+		$page->register_page();
+	}
+
+	/**
+	 * Registers the settings fields for each tab.
+	 */
+	public function register_settings_fields(): void {
+		if ( ! isset( $this->field_collection ) ) {
+			return;
+		}
+		$settings_manager = new Settings_Form_Manager( $this->field_collection );
+		$settings_manager->render_form();
+	}
+
+	/**
+	 * Get the current tab for the settings page.
+	 *
+	 * @param array<string, \WPGraphQL\Logging\Admin\Settings\Fields\Tab\Settings_Tab_Interface> $tabs Optional. The available tabs. If not provided, uses the instance tabs.
+	 *
+	 * @return string The current tab slug.
+	 */
+	public function get_current_tab( array $tabs = [] ): string {
+		$tabs = $this->get_tabs( $tabs );
+		if ( empty( $tabs ) ) {
+			return 'basic_configuration';
+		}
+		// phpcs:ignore WordPress.Security.NonceVerification.Recommended -- Reading GET parameter for tab navigation only, no form processing
+		if ( ! isset( $_GET['tab'] ) || ! is_string( $_GET['tab'] ) ) {
+			return 'basic_configuration';
+		}
+
+		// phpcs:ignore WordPress.Security.NonceVerification.Recommended -- Reading GET parameter for tab navigation only, no form processing
+		$tab = sanitize_text_field( $_GET['tab'] );
+
+		if ( ! is_string( $tab ) || '' === $tab ) {
+			return 'basic_configuration';
+		}
+
+		if ( array_key_exists( $tab, $tabs ) ) {
+			return $tab;
+		}
+
+		return 'basic_configuration';
+	}
+
+	/**
+	 * Load scripts and styles for the admin page.
+	 *
+	 * @param string $hook_suffix The current admin page hook suffix.
+	 */
+	public function load_scripts_styles( string $hook_suffix ): void {
+		// Only load on our settings page.
+		if ( ! str_contains( $hook_suffix, self::PLUGIN_MENU_SLUG ) ) {
```

Comment:
str_contains() is 8.0, consider using strpos() for 7.4 if support is needed.

Reply:
Thanks @josephfusco 

The plugin intentionally is for PHP 8.1 and above but maybe we should put a notice for PHP 7 users.